apps/admin-gui/src/app/shared/components/dialogs/add-role-dialog/add-role-dialog.component.ts

```diff
@@ -74,11 +75,19 @@ export class AddRoleDialogComponent implements OnInit {
     if (this.rules.some((rule) => rule.primaryObject === 'Vo')) {
       this.voService.getMyVos().subscribe({ next: (vos) => (this.vos = vos) });
     }
-    this.groupService
-      .getAllGroupsFromAllVos()
-      .subscribe({ next: (groups) => (this.groups = groups) });
+    this.groupService.getAllGroupsFromAllVos().subscribe({
+      next: (groups) => {
+        this.groups = groups;
+        this.loading = false;
+      },
+      error: () => (this.loading = false),
+    });
     this.resourceService.getAllResources().subscribe({
-      next: (resources) => (this.resources = resources),
+      next: (resources) => {
+        this.resources = resources;
+        this.loading = false;
+      },
+      error: () => (this.loading = false),
```

Comment:
The API calls have to be nested into each other in order for loading to correctly end when the last call ends, no?

Reply:
Instead of basic subscription nesting, I reworked it with rxjs `combineLatestWith` function.